cli/src/commands/git/push.rs

```diff
@@ -375,13 +376,46 @@ pub fn cmd_git_push(
         branch_updates: bookmark_updates,
     };
     let git_settings = tx.settings().git_settings()?;
-    with_remote_git_callbacks(ui, |cb| {
+    let push_stats = with_remote_git_callbacks(ui, |cb| {
         git::push_branches(tx.repo_mut(), &git_settings, remote, &targets, cb)
     })?;
+    process_push_stats(&push_stats)?;
     tx.finish(ui, tx_description)?;
     Ok(())
 }
 
+fn process_push_stats(push_stats: &GitPushStats) -> Result<(), CommandError> {
+    if !push_stats.all_ok() {
+        let mut error = user_error("Failed to push some bookmarks:");
+        if !push_stats.rejected.is_empty() {
+            error.add_formatted_hint_with(|formatter| {
+                for reference in &push_stats.rejected {
+                    write!(formatter.labeled("reference"), "{reference}")?;
```

Comment:
nit: `reference` -> `git_ref` (there's no "reference" label defined in colors.toml)